### DISCUSSION TITLE
Support `alias` for articles

### DISCUSSION THREAD

**Discussion by stereobooster** [withastro/starlight#1847] [Feature Requests]
### What version of `starlight` are you using?

0.22.2

### What is your idea?

Add `alias` [front-matter field](https://starlight.astro.build/reference/frontmatter/#frontmatter-fields). It should be array of strings. For each item it will generate new HTML page, which would have:

- canonical URL of the original page
- meta redirect to the original page
- Maybe JS redirct to the original page
- would be excluded from the sitemap

It should be doable - iterate over all pages, extract all `aliases`, generate new pages.

### Why is this feature necessary?

It is convenient to be able to move pages and not have broken links (from external sites) for the old location

### Do you have examples of this feature in other projects?

- https://gohugo.io/methods/page/aliases/
- https://github.com/HiDeoo/starlight-obsidian/blob/9ba9228b7b75fff5631df70a25914d2d9c4136a6/packages/starlight-obsidian/libs/obsidian.ts#L23-L26

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Astro has a `redirects` feature designed to make it easier to move pages without breaking links: https://docs.astro.build/en/guides/routing/#redirects

Maybe that is sufficient?

**Reply by stereobooster** [reply]
This is possible workaround. But for some reason it seems that keeping it in front-matter is more convenient (not sure why)

**Reply by HiDeoo** [reply]
As you mentioned `starlight-obsidian`, just mentioning that a similar approach could also exist as a plugin of some sort as I basically just read the frontmatter for potential `aliases` and use an [approach](https://github.com/HiDeoo/starlight-obsidian/blob/9ba9228b7b75fff5631df70a25914d2d9c4136a6/packages/starlight-obsidian/libs/starlight.ts#L242-L259) similar to Astro for `redirects`.
